src/main/c/ssl.c

```diff
@@ -231,26 +231,35 @@ static const jint supported_ssl_opts = 0
 
 static int ssl_tmp_key_init_rsa(int bits, int idx)
 {
-#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(OPENSSL_USE_DEPRECATED)
-    if (!(SSL_temp_keys[idx] =
-          RSA_generate_key(bits, RSA_F4, NULL, NULL))) {
-#ifdef OPENSSL_FIPS
-        /**
-         * With FIPS mode short RSA keys cannot be
-         * generated.
-         */
-        if (bits < 1024)
-            return 0;
-        else
-#endif
-        return 1;
-    }
-    else {
+#if defined(OPENSSL_FIPS)
```

Comment:
This Netty code is in the first commit (5404025e0e675bbd4762ef2e0636985532272f53). Grepping for `1024` in OpenSSL's `crypto/rsa` only shows that it's the default size so I'm not actually sure where this 1024-bit limit comes from in the OpenSSL code.